lib/util/ignored-paths.js

```diff
@@ -328,12 +328,16 @@ class IgnoredPaths {
      * @returns {boolean} true if the file path matches one or more patterns, false otherwise
      */
     contains(filepath, category) {
+        const isDir = filepath.slice(-1) === path.sep;
```

Comment:
I like `.endsWith()` 😃 

```suggestion
        const isDir = filepath.endsWith(path.sep);
```

Reply:
Ah, also please check both `/` and `\`. [`CLIEngine#isPathIgnored()`](https://github.com/eslint/eslint/blob/aae6f6525557ba06e73f051511646056313fcf91/lib/cli-engine.js#L900-L904) public API can accept both style paths.

Reply:
@mysticatea I am not sure I understand the last comment correctly, does that mean that I should modify this condition to mark it as a directory if the path ends with either `/` or `\`?
Wouldn't that break on filenames that end with `\` on OSes that use `/` as a path separator?

Reply:
My intention was:

```suggestion
        const isDir = filepath.endsWith("/") || filepath.endsWith("\\");
```

Adding trailing slash is fine with `path.sep` because the path is normalized by `path.resolve()`.

Reply:
@mysticatea, then I have understood you correctly, but I am still concerned about the following situation:
A user of a system with `/` as a path separator has a file named `./some-name\` (where `\` is a part of the filename) and had chosen to ignore directories named that way, so they put `some-name\/` to their ignore file. If we check for both `/` and `\` at the end to determine if something is a directory, both file `some-name\` and directory `some-name\/` will be passed to the `contains()` method as `some-name\/`, which will lead to ignoring both a file and a directory even though only the directory name was specified in the ignore file.
I understand that this is an extremely unlikely situation, but I just want to make sure that this behavior is okay.

Reply:
Actually, I don't have knowledge about *NIX file system. But, at least, Windows allows both `/` and `\` as the path separator and we, Windows users, have often used `/`.